### PR TITLE
Add the release note for C++ client 3.3.0

### PR DESCRIPTION
**Pull request by BewareMyPower**
image

image

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by shibd — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+* [feat][consumer] Support parse broker metadata by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/276
+* add cmake-build-debug dir to gitignore by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/275
+* [docs] Fix the developer guide for macOS users by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/279
+* Fix flaky testConsumerEventWithoutPartition cause by the change of Pulsar 3.0 by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/281
+* [feat][Message] Add getIndex method on Message by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/277
+* Fix testSchemaIncompatibility for Protobuf 3.20.0 or later by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/282
+* Remove log4cxx dependency and its specified configuration by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/283
+* Close the socket gracefully on Windows by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/284
+* Fix ci failed on macos env. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/288
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:24]:
```suggestion
* Fix ci failed on macOS env. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/288
```

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+## What's Changed
+* [feat] Support end-to-end encryption in C Reader API by @rbarbey in https://github.com/apache/pulsar-client-cpp/pull/262
+* [doc] Fix compile perf on MacOS and Ubuntu. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/263
+* [Doc] Add links to client docs and feature matrix in README.md by @momo-jun in https://github.com/apache/pulsar-client-cpp/pull/264
+* Fix deadlock for negative acknowledgment by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/266
+* Support specifying the C++ standard for some higher dependencies by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/269
+* Bumped version to 3.3.0-pre by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/271
+* Fix batch receive for C client unit test faied. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/272
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:14]:
```suggestion
* Fix batch receive for C client unit test failed. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/272
```

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* [improve] AuthAthenz supports Copper Argos by @shustsud in https://github.com/apache/pulsar-client-cpp/pull/274
+* [feat][consumer] Support parse broker metadata by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/276
+* add cmake-build-debug dir to gitignore by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/275
+* [docs] Fix the developer guide for macOS users by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/279
+* Fix flaky testConsumerEventWithoutPartition cause by the change of Pulsar 3.0 by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/281
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:19]:
```suggestion
* Fix flaky testConsumerEventWithoutPartition caused by the change of Pulsar 3.0 by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/281
```

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+* [feat][Message] Add getIndex method on Message by @Shoothzj in https://github.com/apache/pulsar-client-cpp/pull/277
+* Fix testSchemaIncompatibility for Protobuf 3.20.0 or later by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/282
+* Remove log4cxx dependency and its specified configuration by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/283
+* Close the socket gracefully on Windows by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/284
+* Fix ci failed on macos env. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/288
+* Fix broker return error code confuse when not set subscription name. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/289
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:25]:
```suggestion
* Fix broker return error code confusion when not setting subscription name. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/289
```

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* feat: Support message copy for C client. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/285
+* Add missing tls related method definitions for C client. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/287
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:27]:
```suggestion
* Add missing TLS-related method definitions for C client. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/287
```

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+* [feat] Support configure startMessageIdInclusive for the reader by @RobertIndie in https://github.com/apache/pulsar-client-cpp/pull/291
+* Fix build failure with the Protobuf 23.3 by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/290
+* Fix retriable errors not handled well when creating producer or consumer by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/293
+* [fix] producer do not create when topic update partition by @TakaHiR07 in https://github.com/apache/pulsar-client-cpp/pull/295
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:31]:
```suggestion
* [fix] producer does not create when topic update partition by @TakaHiR07 in https://github.com/apache/pulsar-client-cpp/pull/295
```

does not create what?

File: release-notes/versioned/client-cpp-3.3.0.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* Fix the buggy Future and Promise implementations by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/299
+* Fix the test script does not work for Docker on macOS by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/297
+* feat: Support table view for C client. by @shibd in https://github.com/apache/pulsar-client-cpp/pull/294
+* Fix the wrong backoff computation when retrying by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/296
+* Fix the build failure with C++20 standard by @BewareMyPower in https://github.com/apache/pulsar-client-cpp/pull/302
+* [fix] callback of send batch message is error when flush by @TakaHiR07 in https://github.com/apache/pulsar-client-cpp/pull/303
```
Comment (Anonymitaet) [inline release-notes/versioned/client-cpp-3.3.0.md:37]: Sorry, this sentence is incomplete. What's the meaning of "callback of send batch message is error when flush"?

File: release-notes/versioned/client-cpp-3.3.0.md (hunk 6 above)
Comment (BewareMyPower) [reply release-notes/versioned/client-cpp-3.3.0.md:31]: I think it should be "producer won't be created"

File: release-notes/versioned/client-cpp-3.3.0.md (hunk 7 above)
Comment (BewareMyPower) [reply release-notes/versioned/client-cpp-3.3.0.md:37]: I think it should be "[fix] callback of send batch message receives a wrong result when flushing"

**Comment by BewareMyPower** [comment]
@Anonymitaet I've addressed the comments, PTAL.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**